File: PRE_SEASON/Bootcamp_js/my_spaceship/ex00/my_spaceship.js
function my_spaceship(flight_path) {
  //Declaring default variables  
  let x = 0;
  let y = 0;
  let direction = "up";

  //Looping through each character on the flight path to get the coordinate
  for (const character of flight_path) {
    if (character.toLowerCase() === "a" && direction === "up") {
      y--;
    }
    if (character.toLowerCase() === "a" && direction === "down") {
      y++;
    }
    if (character.toLowerCase() === "a" && direction === "left") {
      x--;
    }
    if (character.toLowerCase() === "a" && direction === "right") {
      x++;
    }
    if (character.toLowerCase() === "r" && direction === "up") {
      direction = "right";
      continue;
    }
    if (character.toLowerCase() === "r" && direction === "right") {
      direction = "down";
      continue;
    }
    if (character.toLowerCase() === "r" && direction === "down") {
      direction = "left";
      continue;
    }
    if (character.toLowerCase() === "r" && direction === "left") {
      direction = "up";
      continue;
    }
    if (character.toLowerCase() === "l" && direction === "up") {
      direction = "left";
      continue;
    }
    if (character.toLowerCase() === "l" && direction === "right") {
      direction = "up";
      continue;
    }
    if (character.toLowerCase() === "l" && direction === "down") {
      direction = "right";
      continue;
    }
    if (character.toLowerCase() === "l" && direction === "left") {
      direction = "down";
      continue;
    }
  }

  //returning the resultant values
  return `{x: ${x}, y: ${y}, direction: '${direction}'}`;
}

File: SEASON_1_ARC_1/BOOTCAMP_C/quest04/ex05/my_spaceship.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* my_spaceship (char* paths) {

  int x = 0;
  int y = 0;

  char* direction = "up";

  char* return_result = malloc(sizeof(char)*500); 

    for (int i = 0; i < strlen(paths); i++) {
      
      char path = paths[i];

      
        if(path == 'R'){ 
          if(strcmp(direction, "down") == 0) {direction = "left"; continue; }
          if(strcmp(direction, "left") == 0) {direction = "up"; continue;}
          if(strcmp(direction, "up") == 0) {direction = "right"; continue;}
          if(strcmp(direction, "right") == 0) {direction = "down";  continue;}
        }
        
        if(path == 'L'){ 
            if (strcmp(direction, "right") == 0) { direction = "up"; continue;}
            if(strcmp(direction, "down") == 0) {direction = "right"; continue;}
            if(strcmp(direction, "up") == 0) {direction = "left"; continue;}
            if(strcmp(direction, "left") == 0) {direction = "down"; continue;}
        }
        
        if(path == 'A') {
            if(strcmp(direction, "up") == 0) y--;
            if(strcmp(direction, "down") == 0) y++;
            if(strcmp(direction, "left") == 0) x--; 
            if(strcmp(direction, "right") == 0) x++; 
        }
  
    }


    sprintf(return_result, "{x: %d, y: %d, direction: '%s'}", x, y, direction );

    return return_result;
}
  
